### ISSUE TITLE
ToolStrip overflow menu is visible in UI Automation tree in Content View

### ISSUE THREAD

**Issue by v-elnovikova** [dotnet/winforms#9384]
### .NET version

8.0.100-preview.4.23260.5

### Did it work in .NET Framework?

Yes

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

No

### Issue description

ToolStrip can display some of its items in an overflow menu, when ToolStrip.Size is not big enough, or item has `ToolStripItem.Overflow` set to `Always`. This menu is opened under a 'ToolBar Options' menu item.

According to [MenuItem UIA tree structure](https://learn.microsoft.com/en-us/dotnet/framework/ui-automation/ui-automation-support-for-the-menuitem-control-type#required-ui-automation-tree-structure) overflow menu should be absent from UIA tree in content view:
> For the content view, Menu is absent from the UI Automation tree because it does not convey meaningful information to the end user.

**Actual result**
Menu is visible in content view:
![before](https://github.com/dotnet/winforms/assets/113603457/033735f0-12bb-430b-9623-5429d4bdc7f3)

### Steps to reproduce

Sample app: [WinFormsApp1.zip](https://github.com/dotnet/winforms/files/11895376/WinFormsApp1.zip)

**Comment by John-Qiao** [comment]
Verified this issue in the latest .NET 8.0 SDK build: 8.0.100-preview.7.23356.16, the Menu is still visible in content view, I think that's not the expected result, please check below screenshot:

![Testresult2](https://github.com/dotnet/winforms/assets/45864985/017f9fb5-6d71-402a-bc4c-04bc8671162a)

Menu is visible in control view:
![Testresult1](https://github.com/dotnet/winforms/assets/45864985/8cb2377c-3a23-4c33-a4d6-d2d649556681)

**Comment by v-elnovikova** [comment]
@John-Qiao I cannot reproduce it. Could you please provide the test app that you are running?

**Comment by John-Qiao** [comment]
@v-elnovikova I use the test app [WinFormsApp1.zip](https://github.com/dotnet/winforms/files/11895376/WinFormsApp1.zip) which provided in your reproduce steps. Please check the below gif:

![9384-testresult](https://github.com/dotnet/winforms/assets/45864985/2d5f5e89-ffe2-49f6-a2e4-fe8aa3323229)

**Comment by John-Qiao** [comment]
Verified this issue in the latest .NET 8.0.100-preview.7.23363.19, select the Content option in Accessibility Insights tool, then retest the test app again, the issue is fixed that: Menu option is removed in content view now.
![Testresult4](https://github.com/dotnet/winforms/assets/45864985/1a9a1b36-c881-44c1-9223-7fd761375941)
![Testresult3](https://github.com/dotnet/winforms/assets/45864985/8c0f5a03-01c5-4853-9d03-651d49a8c995)

**Comment by Ashley-Li** [comment]
Verified with .NET SDK 8.0.100-preview.7.23376.3 test pass build, this issue was fixed with above same verification result.
